packages/hash/frontend/src/blocks/page/create-format-plugins/marks-tooltip.tsx

```diff
@@ -34,88 +73,96 @@ export const MarksTooltip: FunctionComponent<MarksTooltipProps> = ({
+        left: "50%",
+        marginTop: -0.5,
         transform: "translate(-50%, -100%)",
         zIndex: "10",
       }}
     >
-      <div
-        style={{
-          display: "flex",
-          backgroundColor: "#ffffff",
-        }}
+      <Stack
+        direction="row"
+        sx={({ palette }) => ({
+          overflow: "hidden",
+          border: `1px solid ${palette.gray[20]}`,
+          borderRightWidth: 0,
+          borderTopLeftRadius: 4,
+          borderBottomLeftRadius: 4,
+        })}
       >
-        {marks.map(({ name, text }) => (
-          <button
-            style={{
-              alignItems: "center",
-              borderColor: "#D1D5DB",
-              borderWidth: "0",
-              cursor: "pointer",
-              display: "flex",
-              paddingBottom: "0.25rem",
-              paddingLeft: "1rem",
-              paddingRight: "1rem",
-              paddingTop: "0.25rem",
-              ...getMarkBtnStyle(name),
+        <FormatButton
+          value="link"
+          selected={activeMarks.some((mark) => mark.name === "link")}
+          onClick={() => {
+            openLinkModal();
+            focusEditorView();
+          }}
+        >
+          <LinkIcon
+            sx={{
+              fill: "inherit",
+              fontSize: 16,
+              mr: 0.5,
+            }}
+          />
+          Link
+        </FormatButton>
+      </Stack>
+
+      <Divider
+        color={theme.palette.gray[30]}
+        sx={{ width: "1px", color: "red" }}
+      />
+
+      <Stack
+        direction="row"
+        sx={({ palette }) => ({
+          p: 0.125,
+          gap: 0.125,
+          overflow: "hidden",
+          border: `1px solid ${palette.gray[20]}`,
+          borderLeftWidth: 0,
+          borderTopRightRadius: 4,
+          borderBottomRightRadius: 4,
+        })}
+      >
+        {marks.map(({ name, icon }) => (
+          <FormatButton
+            value={name}
+            sx={{
+              px: 1,
+              height: 32,
+              borderRadius: 0.5,
+              fill: theme.palette.gray[80],
+
+              [`&.${toggleButtonClasses.selected}`]: {
+                fill: theme.palette.gray[90],
+              },
             }}
             key={name}
-            onClick={() => handleToggleMark(name)}
-            type="button"
+            onClick={() => {
+              toggleMark(name);
+              focusEditorView();
+            }}
+            selected={activeMarks.some((mark) => mark.name === name)}
           >
-            {text}
-            {name === "link" && (
-              <FontAwesomeIcon
-                icon={faCaretDown}
-                style={{
-                  marginLeft: "0.5rem",
-                  fontSize: 10,
-                }}
-              />
-            )}
-          </button>
+            {cloneElement(icon, {
+              sx: {
+                fill: "inherit",
+                fontSize: 16,
+              },
+            })}
```

Comment:
Why not just redefine the definition of `marks` to be:
```typescript

const marks = [
  {
    name: "strong",
    Icon: BoldIcon,
  },
  ...
  {
    name: "highlighted",
    Icon: HighlighterIcon,
  },
];
```
So that you can just do:
```suggestion
            <Icon sx={{ fill: "inherit", fontSize: 16 }} />
```


Is there a benefit to using `cloneElement`?

Reply:
That's a cleaner solution, thanks! Changed in e9a9b3be5582f8d5ded537a8389002e4e1e16751.